src/Peering/Peering/Direct/NewAzureDirectPeeringConnectionCommand.cs

```diff
@@ -0,0 +1,204 @@
+             Mandatory = true,
+
+             HelpMessage = Constants.BandwidthHelp,
+             ParameterSetName = Constants.ParameterSetNameIPv4Prefix + Constants.ParameterSetNameIPv6Prefix),
+         PSArgumentCompleter("10000", "20000", "30000", "40000", "50000", "60000", "70000", "80000", "90000", "100000"),
+         ValidateRange(Constants.MinRange, Constants.MaxRange), ValidateNotNullOrEmpty]
+        public virtual int? BandwidthInMbps { get; set; }
+
+        /// <summary>
+        /// Gets or sets the session ipv4.
+        /// </summary>
+        [Parameter(
+            Position = Constants.PositionPeeringTwo,
+            Mandatory = true,
+
+            HelpMessage = Constants.HelpSessionIPv4Prefix,
+            ParameterSetName = Constants.ParameterSetNameIPv4Prefix)]
+        [Parameter(
+            Position = Constants.PositionPeeringTwo,
+            Mandatory = true,
+
+            HelpMessage = Constants.HelpSessionIPv4Prefix,
+            ParameterSetName = Constants.ParameterSetNameIPv4Prefix + Constants.ParameterSetNameIPv6Prefix)]
+        [ValidateNotNullOrEmpty]
+        public virtual string SessionPrefixV4 { get; set; }
+
+        /// <summary>
+        /// Gets or sets the session ipv6.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+
+            HelpMessage = Constants.HelpSessionIPv6Prefix,
+            ParameterSetName = Constants.ParameterSetNameIPv6Prefix)]
+        [Parameter(
+            Mandatory = true,
+
+            HelpMessage = Constants.HelpSessionIPv6Prefix,
+            ParameterSetName = Constants.ParameterSetNameIPv4Prefix + Constants.ParameterSetNameIPv6Prefix)]
+        public virtual string SessionPrefixV6 { get; set; }
+
+        /// <summary>
+        /// Gets or sets the max prefixes advertised ipv4.
+        /// </summary>
+        [Parameter(
+             Mandatory = true,
+
+             HelpMessage = Constants.HelpMaxAdvertisedIPv4,
+             ParameterSetName = Constants.ParameterSetNameIPv4Prefix),
+         Parameter(
+             Mandatory = true,
+
+             HelpMessage = Constants.HelpMaxAdvertisedIPv4,
+             ParameterSetName = Constants.ParameterSetNameIPv4Prefix + Constants.ParameterSetNameIPv6Prefix),
+         ValidateRange(1, 20000)]
+        public virtual int? MaxPrefixesAdvertisedIPv4 { get; set; }
+
+        /// <summary>
+        /// Gets or sets the max prefixes advertised ipv6.
+        /// </summary>
+        [Parameter(
+             Mandatory = true,
+
+             HelpMessage = Constants.HelpMaxAdvertisedIPv6,
+             ParameterSetName = Constants.ParameterSetNameIPv6Prefix),
+         Parameter(
+             Mandatory = true,
+
+             HelpMessage = Constants.HelpMaxAdvertisedIPv6,
+             ParameterSetName = Constants.ParameterSetNameIPv4Prefix + Constants.ParameterSetNameIPv6Prefix),
+         ValidateRange(1, 2000)]
+        public virtual int? MaxPrefixesAdvertisedIPv6 { get; set; }
```

Comment:
Is there a reason all of these are virtual? If not, please revert.

Reply:
reverted.